features/netsocket/nsapi_dns.cpp

```diff
@@ -1021,7 +1021,15 @@ static void nsapi_dns_query_async_send(void *ptr)
         err = query->socket->sendto(dns_addr, packet, len);
 
         if (err < 0) {
-            query->dns_server++;
+            if (err == NSAPI_ERROR_WOULD_BLOCK) {
+                printf("NSAPI_ERROR_WOULD_BLOCK in nsapi_dns_query_async_send\n");
```

Comment:
Can you at least change this to mbed trace, so that by default it is disabled?

Reply:
Done. 
Sorry for the delay, I was hoping we can test it on the problematic architectures, that should trigger this, before committing, but we can't, as the cellular modems are not available in raas.